### PR TITLE
Pull Request for review

### PR DESCRIPTION
**Pull request by visva-dev**
## Microverse 7th project - Using Bootstrap
- This project requests students to build a replica of the news site Newsweek.com using the Bootstrap framework.
- Aligned all elements with Bootstrap and CSS.

### CODE REVIEW

**Review by BrittanyBlake — CHANGES_REQUESTED** [review]
### Status: Changes Required  🏗️ 🔨 
****
### Hello @visva-dev 👋 
You are doing a great job with this project! Your project looks amazing and you have done a wonderful job implementing the bootstrap classes!🎖️ 🌟 You are almost done. 💪  There are just a few issues that need to be resolved before you can move on: ⬇️ 
****
### Navigation Bar

- [ ] The red background in the nav bar should extend to the width of the whole page.
The original page 👇 
Screen Shot 2020-06-16 at 8 12 43 PM
Your clone 👇 
Screen Shot 2020-06-16 at 8 04 56 PM

- [ ] The search icon in the nav links section is not aligned properly. There is too much space between the search and the icon. Please compare your clone with the original page and fix this. 
Your clone 👇 
Screen Shot 2020-06-16 at 8 10 32 PM
The original page 👇 
Screen Shot 2020-06-16 at 8 10 39 PM



- [ ] On the medium screen, the navigation links should disappear. Please compare your clone with the original page and fix this. 
Your clone 👇 
Screen Shot 2020-06-16 at 8 05 49 PM
The original page 👇 
Screen Shot 2020-06-16 at 8 05 57 PM

- [ ]The same should apply for the small screen. Please see the images below
Your clone 👇 
Screen Shot 2020-06-16 at 8 06 10 PM
The original page 👇 
Screen Shot 2020-06-16 at 8 06 17 PM

### Main content

- [ ] There is a horizontal scroll on the medium screen and too much white space on the right side of your clone. Please fix this.
Screen Shot 2020-06-16 at 8 30 21 PM 1

- [ ] There is also a horizontal scroll and extra white space on the right side on the small screen size. Please fix this. 
Screen Shot 2020-06-16 at 8 33 03 PM

- [ ] On the medium screen of your clone, there is too much space between the images and the text on the opinion and the debate section. Please compare your clone with the original page and fix this.
Your clone 👇 
Screen Shot 2020-06-16 at 8 35 38 PM
The original page 👇 
Screen Shot 2020-06-16 at 8 35 47 PM

- [ ] On the small screen size, the images and text in the opinion and debate section should remain in one line. Please compare your clone with the original page and fix this.
The original page 👇 
Screen Shot 2020-06-16 at 8 45 30 PM
Your clone 👇 
Screen Shot 2020-06-16 at 8 45 37 PM

- [ ] The articles in the "In the Magazine" section on the medium screen are currently occupying too much of the width causing a horizontal scroll. Additionally, the images are stretched in the medium screen and the title below the images should not wrap to a new line. Please compare your clone with the original page and fix this.
Your clone 👇 
Screen Shot 2020-06-16 at 8 49 22 PM
The original page 👇 
Screen Shot 2020-06-16 at 8 49 28 PM

- [ ] The image tag `u.s.` on the article images in the "Editor's Pick" section on the medium and small screen sizes should remain in the bottom left corner of the image as it is on the large screen size. Please compare your clone with the original page and fix this. 

The original page 👇 
Screen Shot 2020-06-16 at 8 54 29 PM
Your clone 👇 
Screen Shot 2020-06-16 at 8 54 24 PM

- [ ] There is too much space between the side images and the text in the Featured slideshows section. Additionally, the side images should not be rounded.  
Your clone 👇 
Screen Shot 2020-06-16 at 8 59 04 PM
The original page 👇 
Screen Shot 2020-06-16 at 8 59 10 PM

- [ ] On the small screen of the Featured slideshows section, the side images and text should be aligned in one line. The text should not wrap to the other line. Please compare your page with the original and fix this. 
The original page 👇 
Screen Shot 2020-06-16 at 9 02 23 PM
Your clone 👇 
Screen Shot 2020-06-16 at 9 02 35 PM

- [ ] The header for the subscribe and newsletter sections are missing.
- [ ] The newsletter on the medium screen size is not formatted properly. Please compare your clone with the original page and fix this. 
Your clone 👇 
Screen Shot 2020-06-16 at 9 05 04 PM
Screen Shot 2020-06-16 at 9 05 12 PM
The original page 👇 
Screen Shot 2020-06-16 at 9 05 18 PM

- [ ] The images in the subscribe section are not formatted properly on the small screen. 
The original page 👇 
Screen Shot 2020-06-16 at 9 08 06 PM
Your clone 👇 
Screen Shot 2020-06-16 at 9 08 13 PM
Screen Shot 2020-06-16 at 9 08 23 PM

  - The two magazine covers should overlap

  - The magazine cover, text, and subscribe button are being partially cut off.

- [ ] There should be no image for the newsletter on the small screen. Please compare your clone with the original page and fix this. 
The original page 👇 
Screen Shot 2020-06-16 at 9 13 07 PM
Your clone 👇 
Screen Shot 2020-06-16 at 9 13 14 PM
### Footer

- [ ] The social media icons and the copyright text on the small screen are not properly aligned. Additionally, the text in the Editions sections extends beyond the width of the page. Please compare your clone with the original page and fix these issues.
Your clone 👇 
Screen Shot 2020-06-16 at 9 15 02 PM
The original page 👇 
Screen Shot 2020-06-16 at 9 15 11 PM
### General

- [ ] Please ensure that you are not overusing `!important` in your CSS file. You have many instances of this in your CSS file.

 





















****
### Kindly make the required changes and resubmit for a new code review once you are ready 🙏


Good Luck and Happy Coding 🍀 👩🏾‍💻

_P.S. If you have any questions about my review or need help, you can message me on slack (@ Brittany Blake)._

File: styles.css [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  margin-top: -15px;
+}
+
+.see-all {
+  margin-top: -1.3rem !important;
+  margin-left: -1.3rem !important;
+  margin-right: -1.3rem !important;
+  padding: 1rem;
+  margin-bottom: -5px !important;
+}
```
Comment (BrittanyBlake) [inline styles.css]: Please try to minimize use of `!important`

File: styles.css [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+.periscope-culture {
+  margin-top: 1.5rem;
+  margin-left: -0.6rem !important;
+  margin-right: -0.6rem !important;
```
Comment (BrittanyBlake) [inline styles.css]: Please try to minimize use of `!important`

File: styles.css [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+}
+
+.editors-pick {
+  margin-right: -1.5rem !important;
+}
+
+.editors2 {
+  margin-right: -3rem !important;
+  margin-top: 2rem !important;
+}
```
Comment (BrittanyBlake) [inline styles.css]: Please try to minimize use of `!important`

File: styles.css [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  left: -120px;
+  top: 18px;
+}
+
+.in-depth {
+  margin-left: -6rem !important;
+}
+
+.text-position {
+  margin-left: -1.1rem !important;
```
Comment (BrittanyBlake) [inline styles.css:64]: Please try to minimize use of `!important`

File: styles.css [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+  .category,
+  .category-2,
+  .category-3,
+  .category-4,
+  .category-5,
+  .category-6,
+  .category-7 {
+    font: 'Roboto Condensed', sans-serif;
+    color: #f72210;
+    text-transform: uppercase;
```
Comment (BrittanyBlake) [inline styles.css]: [OPTIONAL]- You can use the bootstrap class `text-uppercase` for this. Feel free to use [this cheatsheet](https://hackerthemes.com/bootstrap-cheatsheet/) for more information.

File: styles.css [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+.category-6 {
+  bottom: 5.9em;
+}
+
+.vs-img {
+  display: flex;
+  font: bold 13px/20px 'Roboto condensed', Arial, sans-serif;
+  padding: 8px 15px 9px;
+  text-transform: uppercase;
```
Comment (BrittanyBlake) [inline styles.css]: [OPTIONAL]- You can use the bootstrap class `text-uppercase` for this. Feel free to use [this cheatsheet](https://hackerthemes.com/bootstrap-cheatsheet/) for more information.

**Comment by visva-dev** [comment]
- Fixed navbar extends to the whole page.

**Comment by visva-dev** [comment]
## Navigation Bar
- Fixed the navbar extends to the whole page
- Aligned search icon properly in the nav links
- Made nav bottom links to disappear until the large screen
## Main content
- Fixed horizontal scroll on md, lg screen sizes & too much white space problems
- Fixed styling of Debate and Opinion sections on md & sm screens
- Fixed editors section horizontal scroll & images stretch & title wrap on md screen
- Fixed featured slideshows section alignment on sm and md screens sizes
- Updated category styling
- Reduced using of !Important 
## Footer
- Fixed subscribe and newsletter section responsiveness

File: styles.css [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+  .category-7 {
+    font-family: 'Roboto Condensed', sans-serif;
+    bottom: 6.2em;
+  }
+
+  .category-8 {
+    font-family: 'Roboto Condensed', sans-serif;
+    bottom: 6.2em;
+  }
+}
```
Comment (rammazzoti2000) [inline styles.css]: No repeating code, you can notice all this classes are using the same font-family and most of them the same font-size, please refactor that.

File: styles.css [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+  .category-6 {
+    font-family: 'Roboto Condensed', sans-serif;
+    bottom: 6.6em;
+  }
+
+  .category-7 {
+    font-family: 'Roboto Condensed', sans-serif;
+    bottom: 15.2em;
+  }
+}
```
Comment (rammazzoti2000) [inline styles.css]: Same as above

File: styles.css [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+  bottom: 12.2em;
+}
+
+.vs-img {
+  display: flex;
+  font: bold 13px/20px 'Roboto condensed', Arial, sans-serif;
+  padding: 8px 15px 9px;
+  bottom: -20px;
+  z-index: 1;
+}
```
Comment (rammazzoti2000) [inline styles.css]: No repeating code, you can notice all this classes are using the same font-family and most of them the same font-size, please refactor that.

**Review by rammazzoti2000 — CHANGES_REQUESTED** [review]
### Hello @visva-dev :raised_hands: I am Alexandru Bangau, your code reviewer for this project :smile
Great job on this project and of course trying to implement the changes required by the previous TSE :1st_place_medal: .
There are still some issues that need to be fixed before approving this project, see the comments bellow :point_down: 
## :stop_sign: Required Changes :construction: :construction_worker: 
## Large Screen :desktop_computer: 
### Navbar:
- [ ] When resizing the links are not following the width of the container and as you can see from the screenshot the search input gets hidden away
image
- [ ] It seems that the links in navbar are not taking the full width of the header container. As you can notice from the screenshot, there is also some alignment issues with the logo that is not centered and the buttons (login, subscribe) not align to the right.
#### Original:
image
#### Clone:
image
- [ ] The debate and opinion still have too much space between the picture and the title description
#### Original:
image
#### Clone:
image
- [ ] More Stories section doesn't match properly, is missing the article content, you have the author instead.
#### Original:
image
#### Clone: 
image
- [ ] In the `In the Magazine` section you can notice the category labels(periscope/culture) are missing some padding, please add some
image
- [ ] The right side articles on each specfic category is not quite as it supposed to be, you can compare it in the screenshots bellow
#### Original
image
#### Clone
image
- [ ] In the `Subscribe & Culture` section you can see that when resizing the content gets messed up this is usually when the child content doesn't follows the parent container, take a look
image

### Medium Screen
- [ ] The `Navbar` still presents some alignment issues with the `login, subscribe and hamburger menu`, they should properly align to the right
image
- [ ] In `The Debate` section you can see the thumb icon on the right is almost half hidden, see the screenshot bellow
image
- [ ] The picture in the `Culture` section is not showing, while on the original page it does
#### Original
image
#### Clone 
image
### Small Screen
- [ ] You can notice that the navbar still presents some issues, while the `Featured Stories` is the first content showing and the `Top Stories` is placed after it which is actually the wrong order. Also the label `US` should be placed on the bottom left corner
#### Original
image
#### Clone
image
- [ ] You have some sections, like `My Turn` and `Culture and travel` etc, that are taking the full with of the screen, instead they should keep the same margin lines with the rest of the page content. They also have the label not sticking to the bottom left corner where they actually belong to
image
- [ ] Section `The Editor's Pick` and `In the Magazine` are not centered, you can see they align to the left, also the same problem here with the labels not stickink to the right spot
image
- [ ] Here the main article picture is taking again the whole width, while the article bellow are taking only a portion
#### Original
image
#### Clone
image
- [ ] Here the text is overflowing outside the main container
image
 - [ ] The footer is not behaving like the original, see the screenshot below
#### Original
image
#### Clone
image

### :exclamation:  Note: 
Overall the project needs more improvements on the typography, like font-family styling, font-size, color etc.
Also if you compare the original project when hovering over anchor tags it will either be underlined or will change the color, yours it not doing that, not only once, mainly because in many places you didn't use anchor tags but either headers or paragraphs

### <em>Kindly fix the REQUIRED issues and send another code review request.

Cheers and Happy coding! :clap: 

Code review by Alexandru Bangau

(Feel free to contact me on slack by the handle of **_alexandru_** if you have any additional question)</em>
### :stop_sign: Status: Required Changes

**Comment by visva-dev** [comment]
## Issues fixed 
- Fixed bottom-nav styling for all screen sizes
- Fixed featured stories category styling
- Fixed all categories not sticking to the bottom problem
- Fixed More stories section alignment on all screens
- Updated responsiveness of debate and opinion sections
- Fixed right side articles styling
- Menu and subscribe sections updated style
- Fixed footer positioning
- Updated order
- Cleaned syntax with no repeating code approach

**Review by christianotieno — CHANGES_REQUESTED** [review]
# **CHANGES REQUIRED** :warning:
Hey @visva-dev   :wave:
-
You have done a splendid job with the changes! :tada: Keep it up.
We, however, do have some few changes to carry out before we proceed (see review comment).
##

### Final Touches:

- Let's validate the markup HTML using this [tool](https://validator.w3.org/nu/#textarea) to remove the semantic errors.
- Let's update the live link of the page in the README file to reflect the new changes.
-  In the subscribe section, we have an issue. Refer to the screenshots below 👇  for comparison then tweak the small changes.
 
|[Source Page](https://www.newsweek.com/)|[Clone Page](https://raw.githack.com/visva-dev/PROJECT-USING-BOOTSTRAP/feature-branch/index.html)|
|:---:|:---:|
|small-screen|small-screen-clone|

-  Kindly reduce the sizes of the images a bit and centre the texts properly, thank you.

### OPTIONAL CHANGES.
- Improve the general colours of the page. They are currently a bit off. Let's see how they compare below and maybe try and work on that as well.

|[Source Page](https://www.newsweek.com/)|[Clone Page](https://raw.githack.com/visva-dev/PROJECT-USING-BOOTSTRAP/feature-branch/index.html)|
|:---:|:---:|
|source|clone|

- If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email. :eyes:
##
- When you have made the specified changes, please submit another code review. Thank you!
Happy coding :smile:
**[Christian Otieno](https://github.com/ChristianOtieno)**
[Twitter](https://twitter.com/iamchrisotieno) [linkedin](https://www.linkedin.com/in/christianotieno/) [GitLab](https://gitlab.com/christianotieno)
---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

**Comment by visva-dev** [comment]
## Changes made
- Fixed syntax errors
- Updated positioning of subscribe section
- Updated live dome link in the readme file

**Review by christianotieno — APPROVED** [review]
# PROJECT APPROVED :champagne:
***All the required tasks have been fulfilled in this project. Everything looks clean now. I will go ahead and mark this as complete.***
- Congratulations on the splendid job @visva-dev! :+1:
- You can now merge this. I wish you all the very best in your next tasks 🥇 
##
Happy coding :smile:
**[Christian Otieno](https://github.com/ChristianOtieno)**
[Twitter](https://twitter.com/iamchrisotieno) [linkedin](https://www.linkedin.com/in/christianotieno/) [GitLab](https://gitlab.com/christianotieno)

**Pull request merged**